### PR TITLE
test: add tests for workflow run controller

### PR DESCRIPTION
**Pull request by chalindukodikara**
## Purpose
$subject

## Approach
This pull request enhances the `WorkflowRun` controller with improved integration test coverage and error handling, especially around status management, external reference evaluation, and reconciliation lifecycle. It also removes unreachable code paths and adds new unit tests for template-based name evaluation.

**Integration and Unit Test Improvements:**

* Added comprehensive integration tests in `controller_integration_test.go` to cover scenarios such as early exit for completed workflows, multi-cycle reconciliation, workflow resolution failures, persistence of status fields, and validation with `ClusterComponentType`. These tests ensure the controller’s behavior is robust and correct across various edge cases.

* Introduced new unit tests for `evaluateExternalRefName` and error handling in external reference resolution, ensuring template/CEL expression evaluation behaves as expected and errors are surfaced clearly.

**Controller Logic and Error Handling:**

* Removed unreachable code branches in the controller’s reconcile and finalize logic that handled `nil` workflow plane results, as these cases are now handled earlier in the reconciliation process. This simplifies the code and prevents redundant condition setting. [[1]](diffhunk://#diff-074b6c77be9ef38c097ba573ffbefbe7f53066c01511d24a6acc0360f185afcaL167-L172) [[2]](diffhunk://#diff-dd2294ace327b02eae84738f9a39366af58f7fb51a670e7b3ee47be63c3f04fdL73-L76)

**Test Infrastructure:**

* Added import for the `template` package in the external reference test file to support new template evaluation tests.

## Related Issues
https://github.com/openchoreo/openchoreo/issues/3073

## Checklist
- [ ] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)
- [ ] Added `backport/<release-branch>` label if this should be backported (e.g., `backport/release-v1.0`)

### CODE REVIEW

File: internal/controller/workflowrun/controller.go [hunk 1]
```diff
@@ -164,12 +164,6 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		setWorkflowPlaneResolutionFailedCondition(workflowRun, err)
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
-	if workflowPlaneResult == nil {
```
Comment (LakshanSS) [inline internal/controller/workflowrun/controller.go:167]: is the nil case now handled somewhere else?

**Review by LakshanSS — DISMISSED** [review]

File: internal/controller/workflowrun/controller.go (hunk 1 above)
Comment (chalindukodikara) [reply internal/controller/workflowrun/controller.go:167]: This section is dead code. It will never happen.
